sdk/core/azure-core/azure/core/pipeline/_base_async.py

```diff
@@ -47,9 +48,9 @@ class _SansIOAsyncHTTPPolicyRunner(
     :type policy: ~azure.core.pipeline.policies.SansIOHTTPPolicy
     """
 
-    def __init__(self, policy: SansIOHTTPPolicy[HTTPRequestType, AsyncHTTPResponseType]) -> None:
+    def __init__(self, policy: SansIOHTTPPolicy[HTTPRequestType, HTTPResponseType]) -> None:
```

Comment:
I'm quite confused why you replaced a TypeVar with a new TypeVar ? Now the constructor inheritance isn't matching anymore.